pilot/pkg/networking/core/configgen.go

```diff
@@ -37,6 +37,10 @@ type ConfigGenerator interface {
 	// BuildClusters returns the list of clusters for the given proxy. This is the CDS output
 	BuildClusters(node *model.Proxy, push *model.PushContext) ([]*discovery.Resource, model.XdsLogDetails)
 
+	// BuildDeltaClusters returns both a list of resources that need to be pushed for a given proxy and a list of resources
+	// that have been deleted and should be removed from a given proxy. This is Delta CDS output.
+	BuildDeltaClusters(node *model.Proxy, push *model.PushContext, updates *model.PushRequest) ([]*discovery.Resource, []*discovery.Resource, model.XdsLogDetails)
```

Comment:
I don't think returning `[]*discovery.Resource` for deleted is the right approach. There is no need to build the entire cluster just to tell Envoy to remove it . I think it would not work anyways, as if the service is deleted, then `proxy.SidecarScope.Service` will return `nil` and we skip the whole thing.

We probably just want to return the names

Reply:
Yes, the delta xds response only need to know the resource names for the deleted resources

Reply:
Maybe call `DeltaBuildClusters` to be consistent with `DeltaAggregatedResources`

Reply:
> We probably just want to return the names
Got it. I'll return a `[]string`
> Maybe call DeltaBuildClusters to be consistent with DeltaAggregatedResources
sure